src/zope/component/tests/test_globalregistry.py

```diff
@@ -214,7 +219,7 @@ class Foo(object):
             pass
         @adapter(IFoo)
         def _handler(context):
-            assert 0, "DON'T GO HERE"
+            raise NotImplementedError()
```

Comment:
Assertions can be turned off. It's not common under test code (I hope) but it is possible.